tasks/redhat.task/kickstart.erb

```diff
@@ -11,7 +11,20 @@ rootpw <%= node.root_password %>
 network --hostname <%= node.hostname %>
 firewall --enabled --ssh
 authconfig --enableshadow --passalgo=sha512 --enablefingerprint
-timezone --utc America/Los_Angeles
+timezone --utc <%= node.metadata['timezone'] || 'America/Los_Angeles' %>
+rhel_username="<%= node.metadata['rhn_username'] %>"
```

Comment:
Also, there's a command 'ksvalidator' which you can feed a kickstart file and have it check for syntax errors